### PR TITLE
Proposal: ZooKeeper discovery

### PR DESCRIPTION
**Pull request by chanwit**
This PR implements the ZooKeeper discovery service. It's been tested with ZooKeeper 3.3.x and 3.4.x.

### CODE REVIEW

File: discovery/zookeeper/zookeeper.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	var nodes []*discovery.Node
+	for _, addr := range addrs {
+		nodes = append(nodes, discovery.NewNode(addr))
+	}
+
+	return nodes, nil
+}
+
+func (s *ZkDiscoveryService) Watch(callback discovery.WatchCallback) {
+	for _ = range time.Tick(time.Duration(s.heartbeat) * time.Second) {
```
Comment (vieux) [inline discovery/zookeeper/zookeeper.go]: Can you use zookeeper watches instead of a timer here please ?

Like we do in etcd.

**Comment by chanwit** [comment]
Hi @vieux I've implemented a proper watcher.

To confirm that the watcher kicks, Please enable Debug logging  and check a log entry saying:
`[ZK] Watch triggered`.

**Comment by vieux** [comment]
LGTM

**Comment by vieux** [comment]
Thank you!

**Pull request merged**